lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -92,7 +92,9 @@ def group_name
 
     def run_kubectl(*args)
       raise FatalDeploymentError, "Namespace missing for namespaced command" if namespace.blank?
-      args = args.unshift("kubectl").push("--namespace=#{namespace}")
+      raise FatalDeploymentError, "Explicit context is required to run this command" if context.blank?
+      args = args.unshift("kubectl").push("--namespace=#{namespace}").push("--context=#{context}")
```

Comment:
`push` mutates the original array which may introduce unexpected and dangerous side-effects

```ruby
args = ["original_command"]
run_kubectl(args) # you don't suppose that `run_kubectl` mutates the original argument
# but now `args` is mutated and contains three extra array items
```